### DISCUSSION TITLE
New features and bug fixes will occur in this repo

### DISCUSSION THREAD

**Discussion by cbaker6** [netreconlab/Parse-Swift#7] [Announcements]
I will continue to add new features and bug fixes that align with the core design of the Parse Swift SDK in this repo. If anyone would like to know for the reasons, I briefly explain some of them below:

> The reason for the bifurcation from the parse-community version of Parse-Swift is due to interference and a number of disagreements with the [Parse Project Management Committee (PMC)](https://github.com/parse-community/Governance/blob/main/TEAM.md#project-management-committee-pmc) about the future of ParseSwift along with the PMC member ignoring code reviews and comments, marking relevant comments as `off-topic`, merging commits directly to main branch, lack-of-understanding of Swift fundamentals, client-side development, and lack of knowledge about how the Parse-Swift framework is designed. It is important to emphasize that no member of the Parse PMC participated in the design, release, and direction of Parse-Swift from its [first commit](https://github.com/parse-community/Parse-Swift/tree/cf69b7f0638819a7070b82228bc51a97df5757e6) to version [4.14.2](https://github.com/parse-community/Parse-Swift/releases/tag/4.14.2). The Parse PMC also does not properly give attribution or show appreciation to Corey as the lead developer of Parse-Swift. In addition, no funding or support from the [parse-community funds](https://opencollective.com/parse-server) has ever been offered to Corey for any of his [contributions to Parse](https://github.com/search?q=user%3Aparse-community+author%3Acbaker6&type=pullrequests), though a [number of funds have been offered and made to other contributors](https://opencollective.com/parse-server#category-BUDGET).

Parse-Swift<sup>OG</sup> will remain aligned with the original core principals of a swifty framework that contains zero dependencies and takes advantage of all of the features the [parse-server](https://github.com/parse-community/parse-server) has to offer.

Migration from parse-community/Parse-Swift <= [4.14.2](https://github.com/parse-community/Parse-Swift/releases/tag/4.14.2) to Parse-Swift<sup>OG</sup> version [4.16.2](https://github.com/netreconlab/Parse-Swift/releases/tag/4.16.2) is a simple one-line change, see #70 for directions. [Migrating to Parse-Swift<sup>OG</sup> 5.0.0+](https://github.com/netreconlab/Parse-Swift/discussions/74) is a little more involved.

Feel free to open issues and discussions about problems you run into.

Thanks for using Parse-Swift and I look forward to collaborating with you all!

**Comment by cbaker6** [comment]
cc: @pmmlo @vdkdamian @Vortec4800 @rocxteady @ulitiy @bcbeta @gemiren @InstaRobot @asielcabrera @Climbatize

**Comment by vdkdamian** [comment]
I'm not a big fan of this move, but I understand why it happened.
Going your own way with this repo can't be good for the whole Parse community.

**Reply by cbaker6** [reply]
> Going your own way with this repo can't be good for the whole Parse community.

Maybe, the Swift SDK was originally under Flo's direction and then under my direction thereafter. I'll continue the SDK on the course that has been set from its origins. It's open-source, so the community can choose which repo better suites their needs going forward. Everyone is also free to contribute to this repo so we can make continued improvements together.

As for the parse community repo, of course devs are can use it if that makes them feel more comfortable. Individuals from the Parse PMC have been attempting to control its direction since the Swift SDK picked up steam, though none of them (or anyone currently part of the Parse governance) contributed to it's design, development, or direction in the first place. Attempting to force direction and design, without understanding the current system is always a bad sign IMO. I suspect, under the Parse PMC's guidance, that particular repo will end up in a similar state as many of the other client SDK's (iOS, Android, Flutter) and be riddled with issues and pending PR's for simple bug fixes that take forever to get addressed.

**Comment by pmmlo** [comment]
I agree with others but respect the decision.

I think considering @cbaker6 has single-handedly driven this project for the last two years with the majority of functional commits he has the right to take ownership of/credit for it.

I do think it would be of great communal benefit if you do keep it open source, like other examples of your work, but I do think you should ultimately have the choice.

Thanks for all the work you have done.

**Reply by pmmlo** [reply]
Also, appreciate you giving a heads up.

**Reply by cbaker6** [reply]
> Thanks for all the work you have done.

I appreciate you!

> I do think it would be of great communal benefit if you do keep it open source, like other examples of your work, but I do think you should ultimately have the choice.

I'm an academic and the reason why I took on Parse-Swift is to create a SDK that can quickly assist with making new research based apps and systems. I collaborate with academia and industry to build these systems so the all of the projects linked in this repo will remain open-source as they should be able to help anyone who needs them.

**Reply by pmmlo** [reply]
I'm a researcher, raised by an academic.  Share your mission.

**Comment by darwishoo** [comment]
@cbaker6 I am a big fan of your work and would love the chance to contribute to this project in anyway shape or form.

**Reply by cbaker6** [reply]
Thank you! There's definitely a few things that still need to be working on:

- [ ] Adding support for multiple users: https://github.com/parse-community/Parse-Swift/projects/2#card-68137924
- [ ] Converting the current ParseLiveQuery to be protocol based. This will allow developers to use other web socket frameworks to conform to the protocol and use LiveQuery on Linux, Android, and Windows: https://github.com/parse-community/Parse-Swift/projects/2#card-68137924
- [ ] Create a local storage protocol for developers to conform to (this may not be needed as all ParseObject's are `Codable`, allowing developers to use other tools) https://github.com/parse-community/Parse-Swift/projects/2#card-53117524

Of course, if you find any bugs or features missing, feel free to open up an issue or pull request

**Comment by vdkdamian** [comment]
@cbaker6 I see a lot of people are still creating issues on the git of Parse community. This is why I was not a big fan of this move. Is there something more you can do to let people know issues are handled here?

**Reply by cbaker6** [reply]
People posting issues on the other repo is okay. Both repos are public, so I can see any issue there. This repo already has a number of fixes ahead of the other repo, over time the amount of features and bug fixes will increase and so will the popularity. The most we can do is post to let people know about this repo and developers will choose for themselves.

**Reply by vdkdamian** [reply]
Okay, great!

**Comment by cbaker6** [comment]
If you are interested in getting the most up-to-date features and bug fixes for Parse-Swift, I strongly encourage you to **star**, **watch**, and **submit issues and pull requests** to https://github.com/netreconlab/Parse-Swift instead of the Parse-Community Parse-Swift. You can read this [discussion thread](https://github.com/netreconlab/Parse-Swift/discussions/7#discussion-4497222) to learn more about why I chose to stop contributing to Parse-Platform repo of Parse-Swift and why I opted-out of the Parse-Platform team. Currently, the **Parse-Platform Parse-Swift repo is significantly behind this repo in terms of bug fixes and features** and this will only continue. In addition, they are currently in [search of contributors](https://github.com/parse-community/Parse-Swift/issues/432) and even attempting to [pay developers using their bounty system](https://github.com/parse-community/Parse-Swift/issues) for bugs I've already fixed. You can see my contributions to Parse-Swift [here](https://github.com/parse-community/Parse-Swift/graphs/contributors).

Many of you learned about Parse-Swift through either [iOS Dev Weekly 560](https://iosdevweekly.com/issues/560#start) or through the [Swift Package Indexing Episode 5 hosted on Twitter Spaces](https://blog.swiftpackageindex.com/posts/swift-package-indexing-episode-5/). Both of the aforementioned mediums discussed and highlighted my development of Parse-Swift and grew the repo's **stars** to it's current numbers.

Point your project repo's to the [latest release of Parse-Swift](https://github.com/netreconlab/Parse-Swift/releases) using SPM. Feel free to open [discussions](https://github.com/netreconlab/Parse-Swift/discussions) on this repo if you have any questions!

There are also a few other Parse repos I have available that can assist you with your development:

- [ParseServerSwift](https://github.com/netreconlab/parse-server-swift) - Write Parse Cloud Code in Swift using Parse-Swift
- [ParseServerAnyAnalytics](https://github.com/netreconlab/parse-server-any-analytics-adapter) - Analytics adapter to connect Parse Server analytics to any 3rd party analytics tool
- [CarekitSampe-ParseCareKit](https://github.com/netreconlab/CareKitSample-ParseCareKit) - An example application of [CareKit](https://github.com/carekit-apple/CareKit)'s OCKSample synchronizing CareKit data to the Cloud via [ParseCareKit](https://github.com/netreconlab/ParseCareKit)
- [ParseCareKit](https://github.com/netreconlab/ParseCareKit) - Synchronize CareKit 2.1+ data with a parse-server using Parse-Swift
- [SnapCat](https://github.com/netreconlab/SnapCat) - SnapCat is a social media application for posting pictures, comments, and finding friends. SnapCat is designed using SwiftUI and the ParseSwift SDK
- [ParseMigrateKeychain](https://github.com/netreconlab/ParseMigrateKeychain) - A sample app that demonstrates how to migrate an app written with the Parse [Objective-C SDK](https://github.com/parse-community/Parse-SDK-iOS-OSX) to the Parse-Swift SDK

cc: @soneppu @zo59nari @eemailme @zclmoon @fitzgebd2 @drdaz @crotel @jhcloos @cbiggin @halldorg @latne1ec @jlott1 @ShawnBaek @shengchalover @karnlund @sinduke @wongct @denizak @BeiXiao @zero-times @alimadhoun @Viverrinus @IlyesDjari @bennetty74 @periclesm @onepisYa @Kondamon @riccardoch @harreed @JqyModi @coughski @MuratCay @OmarJalil @dadurigon @jeremyspritelyco @clt @harrygallagher4 @turgay2317 @Eyesony @eduardo22i @FWDavide @italotoffolo @wjcn @elGeekalpha @JoeFryer @daSkier @abcross92 @pettersillander @ricardopereira @rain2540

**Reply by cbaker6** [reply]
cc: @calvingit @pmusolino @dmitryshliugaev @anyinfa @shoujiaxin @Layzie @OHeroJ @difoi @ipader @arcangelw @CreatorMetaSky @aplekhanov @gapl @xinGGG @atetlaw @calrizien @vhosune @allenhsu @defagos @KS1019 @BalazsSzamody @hanbum @ArchimboldiMao @andrekandore @davepen @ilyagru @narciszait @rbarrosoSky @jscubida @toastersocks @dorjeozer @paulfreeman @dacoffey @FrancoisObob @kalyberk @dizaraj @InstaRobot @dsantix @Mithileshraj123 @yoliOms @thedan84 @nokonoko @MarSYs1 @AJajx

**Reply by cbaker6** [reply]
cc: @Exey @dehengxu @grago @colrack @jondwillis @gandli @kkla320 @Kalzem @League2EB @arthurtop @HappySwifter @frkyldrm @shinjism @ihomway @jowkame @Marcelo-Apps @lcztt @arata-1972 @Atlands @istoleyocookies @kstefanou52 @ivanmkc @ashish-naik @congncif @kilabyte @alex-x-x-x-x @colatusso @farshadmb @phynet @ShivamBhosale @XingGX @theInternauts @SWereszczynski @justin747 @JARMourato @RinatAbidullin @WayneEld @jsonhu @yageek @avoltaire @Guli-vali @kangwooc

**Reply by cbaker6** [reply]
cc: @chenji1990 @adi-li @Mohamed-AbdulRaouf @sanzaru @davidcruzcs @morelearn1990 @cancelself @FarisAlbalawi @mahmutbedir @bahaa-kallas @DpoH @JonathanWDR @wving5 @han-qiu @AveryWenwenSi @jt9897253 @krupitskas @Xevro @oleksanderkorn @GothStar @bayareahank @kkpan11 @jlcreator @KushalBhanot @brandy2015 @nickfalk @BRETT71 @zhangyida-lab @harry-neodoc @mtyn @suosuopuo @denemmy @Jamonek @jeffreyjackson @m4p @tkersey @mkll @athanasiospap @kylehowells @jfk056 @0x0c

**Reply by cbaker6** [reply]
cc: @newbdez33 @jixuqianxing @TimOliver @steipete @alanrmendez @Conceptx @mac-code @gogl92 @JoshedItUp @Re-Krass @kennic @bstillitano @Nef10 @fernandogmo @gkze @andrey-krukovskiy @HackShitUp @mman @starkdmi @Husseinhj @funkenstrahlen @mugecevik @stjordanis @qingfeng14 @viikufa @omkar-tenkale @santiagoprieto @Surfndez @jinSasaki @Kaitachi @TomWFox @konkontos @MoathOthman @Aldoukfm @sambing @monajafi @hauntednomad @sadhudgp91 @rajeshkumarkarra

**Reply by cbaker6** [reply]
cc: @tanmayb123 @pchelnikov @francocorreasosa @skpaik @Onwa1kenobi @98thpixel @danieltarazona @cmcoffeedev @yingmu52 @madurangae @filmhomage @lmick002 @proactive-solutions @mrtavat @nathantannar4 @zakiso @CJ99 @MobileVet @scinfu @JulienKode @Sadmansamee @mooshee @matheusfrozzi @felnit @Chihpin @FridaySG @johncodeos @DevBakura @churchmediasolutions @pungme @oallouch @otymartin @latne1ec @AbcTM @jaysonng @LilMoke @amsh92 @kenanbylan @Jack-Adam-Mwanzo @richardtop @DaedalusofCrete @Metadr0n @KamandaK9

**Reply by Kondamon** [reply]
👍 @cbaker6, I would recommend that you put a short brief and prominent banner on the very top of the main page of this repo. Just describe what this repo differentiates from the parse-community version including your message from above (I'm a researcher, raised by an academic.) add add some links with background info!

**Reply by theInternauts** [reply]
I looked through the discussion thread and I don't see the reason your commits are not being merged. Commits are sitting there from others, as well.  What about your development direction differs from theirs?

**Reply by cbaker6** [reply]
> I looked through the discussion thread and I don't see the reason your commits are not being merged.

@theInternauts It’s not that my commits are not being merged, I’m simply not committing to the parse-community Parse-Swift repo anymore.

> Commits are sitting there from others, as well.

The [Parse-Swift repo on parse-community](https://github.com/parse-community/Parse-Swift) currently has [no pending PR’s](https://github.com/parse-community/Parse-Swift/pulls). The Parse-Community repo does have a [number of issues](https://github.com/parse-community/Parse-Swift/issues), the issues listed as **bugs** are all fixed in the [netreconlab repo latest release](https://github.com/netreconlab/Parse-Swift/releases). Maybe you are referring to the [Objective-C SDK](https://github.com/parse-community/Parse-SDK-iOS-OSX) which differs from the Swift SDK? If so, this thread is not about that SDK.

> What about your development direction differs from theirs?

My direction is discussed [here](https://github.com/netreconlab/Parse-Swift#use-parse-swift-from-netreconlab) and is based on the [original direction](https://github.com/parse-community/Parse-Swift/issues/3#issuecomment-328351357) set by Flo and myself. The Parse PMC is locked-in on simply forcing the SDK to operate like the other client SDK’s (Objective-C, JS, Android, Flutter, etc.). Though those are powerful, they are based on the archaic design architecture created when Parse was owned by Facebook and pre-dates when the ecosystem was open-sourced (for example, Swift 1 wasn’t released when the original Parse iOS SDK was released, we are now on Swift 5.7). They will eventually force in dependencies, make the code clunky, and restrict the capabilities of Parse-Swift either on-purpose or through lack of understanding of the current design of the Swift SDK. The direction I’ve carried Parse-Swift in has led to the most flexible Parse Client SDK to date, the most feature rich SDK, the only SDK with zero dependencies, 1 of 2 SDK’s that works with Cloud Code (the other is the JS SDK), all of which works on all Apple platforms, Linux, Android, and Windows

**Reply by jaysonng** [reply]
Just wanted to drop in and say I too appreciate all your hard work on Parse-Swift for the community. Thank you very much!

**Comment by dblythy** [comment]
Thank you for your contributions to the OG Parse Swift repo, I’m sorry if you felt unappreciated by our community. I enjoyed our discussions and learnt a lot from the few PRs I wrote with your guidance and assistance

**Reply by cbaker6** [reply]
Thanks @dblythy! I’ve enjoyed your improvements, bug findings, and discussions as well!… Also, wonderful work you’ve been doing on the server, your additions are valuable improvements.

**Reply by cbaker6** [reply]
I do believe the parse-community appreciated my contributions. My gripes are more with 1 PMC member and how blanket authority is asserted without accountability, knowledge, and lack of checks and balances for this member. In addition, this person attempts to act as if they are someone’s “boss”. For example, one time they direct messaged me after a blog came out about Parse-Swift telling me I need to contact and verify with them about what I plan to say in future blogs/interviews as if I needed approval from them. These types of requests are absurd and are a sign of a “power trip”. The PMC consists of many people, yet only the one speaks on behalf of all of them in the last year or so. I enjoyed the PMC when they had Tom and the others interacted with the community and on repos.

**Comment by cbaker6** [comment]
Parse-Swift users, version [5.0.0](https://github.com/netreconlab/Parse-Swift/releases/tag/5.0.0) has finally been released and has a number of amazing updates. To learn more about the changes and how to migrate, please see #74. Examples of how to use 5.0.0+ can be found in the [Swift Playgrounds](https://github.com/netreconlab/Parse-Swift/tree/main/ParseSwift.playground/Pages)

I've also composed some [Feature Comparison Tables](https://github.com/netreconlab/Parse-Swift/discussions/72) which compares Parse-Swift<sup>OG</sup> to parse-community/Parse-Swift, Parse-SDK-iOS-OSX, Parse-SDK-Flutter, and Parse-SDK-JS.

A few other discussions have been added as well:
1. See the https://github.com/netreconlab/Parse-Swift/discussions/74 to learn how to migrate from Parse-Swift<sup>OG</sup> 4.15.0+ to 5.1.1+
1. See the https://github.com/netreconlab/Parse-Swift/discussions/70 to learn how to migrate from [parse-community/Parse-Swift](https://github.com/parse-community/Parse-Swift)
1. See the https://github.com/netreconlab/Parse-Swift/discussions/71 to learn how to migrate from [Parse-SDK-iOS-OSX](https://github.com/parse-community/Parse-SDK-iOS-OSX)

Also remember that Parse-Swift<sup>OG</sup> can be used to write Cloud-Code and skip all of the JS by using [ParseServerSwift](https://github.com/netreconlab/parse-server-swift)

Be sure to **star** and **watch** this repo to keep up with the latest changes. Also, feel free to comment on the linked discussions if you see any fixes or have more info to contribute!

cc: @softcleandev, @Skeeze22, @nikitavoloboev, @EgorVedeneev, @kellvembarbosa @kennic @claesjacobsson @project-academy @Aziz-AnyDesk @wonder2011 @windorz9 @Mater2211 @Neilfau @jiasheng-li @lephnganh @gbfansheng @TaoLyn838 @yunhao @AriestaAgung @esikmalazman @MattDuyv @Phinny01 @skyofdwarf @benwoosley @magnett @bogdanf @Josscii @meterscao @EricKilonzo @MrDevel0per @Marcelo-Costa-Dev @adamhoracek @CmST0us @LeoKlaus @Den1Doc @idewindy

**Reply by softcleandev** [reply]
@cbaker6 I just read through the 5.0.0 changelog. You have done a very good job. Keep it up!

**Comment by cbaker6** [comment]
@nebitrams you should use this repo if you want the latest ParseSwift updates. You can look through the thread above to learn why.

There are additional [discussions](https://github.com/netreconlab/Parse-Swift/discussions) on how to [upgrade/migrate](https://github.com/netreconlab/Parse-Swift/discussions/70#discussion-4952339) and [why this repo is better](https://github.com/netreconlab/Parse-Swift/discussions/72).

Cc: @lexiaoyao20 @ardanar @lexiaoyao20 @maximkhatskevich @Formerly @tobi404 @alessdiimperio @Yita55 @glussian-luna @ValentinoRusconi-EH @enesgumus @lsmilek1 @Tymoteusz24 @Erica-Iris @misteu @PaoloProdossimoLopes @jon-tous @markiv @ellekadfur @Lizet6 @hunterh37 @mcmay @pktealshift @0w0Neko @beraatoztorun @muhammedturabisancak @leonliman @p-Lukas @souvik666 @joeavargas @Coriolan-Bataille @technicat @YassineLafryhi @WSMI052 @Drendroyt @YassineLafryhi @A-Oluc @esummers @davidhakki @Dommgrand @samiguntepe @Rakhimjon @sjmn @unrealfish @RomanPodymov @Stars1233 @MichalTalaga17 @Trust-Worthy @Jenkins1128 @Hephorus @dindacantik220906

**Reply by misteu** [reply]
Thank you for the information! Was not aware. I used the docs from back4app and unfortunately they do point to this old community version.

Thanks again for cc'ing!

**Comment by cbaker6** [comment]
Hey everyone, `ParseSwift 6.0.0` with complete concurrency has been released for `Swift 6.0+`. Learn more about the release here: https://github.com/netreconlab/Parse-Swift/discussions/225

**Reply by cbaker6** [reply]
For those interested in latest releases of ParseSwift, please see the threads above

cc: @Chtapodi0 @GoStraight-Tech @antjumper @cspell2k5 @jacog2 @Branchrise @gut-puncture @harshithsudarsan @dclawd88
